engineering_tools/carma_script_extensions/swarm

```diff
@@ -0,0 +1,352 @@
+#!/bin/bash
+
+#  Copyright (C) 2023-2024 LEIDOS.
+# 
+#  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+#  use this file except in compliance with the License. You may obtain a copy of
+#  the License at
+# 
+#  http://www.apache.org/licenses/LICENSE-2.0
+# 
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#  License for the specific language governing permissions and limitations under
+#  the License.
+
+
+swarm__install() {
+    if [ "$1" = "-d" ]; then
+        local USERNAME=usdotfhwastoldev
+        local TAG=$2
+    elif [ "$1" = "-c" ]; then
+        local USERNAME=usdotfhwastolcandidate
+        local TAG=$2
+    else
+        local USERNAME=usdotfhwastol
+        local TAG=$1
+    fi
+
+    if [[ -z $TAG ]]; then
+        echo "Please specify a tag string to update carma-config-data volume."
```

Comment:
This should print out to stderr, not stdout. This comment applies to several locations.

Reply:
We discussed that stderr is probably best defined as a function in the main carma script to use in extensions instead of defining it in the extension itself. I'll compile a list of edits/enhancements that the primary carma script needs such as this.